benchmark/Streamly/Benchmark/Data/Parser.hs

```diff
@@ -18,21 +19,30 @@ import Data.Foldable (asum)
 import Data.Functor (($>))
 import Data.Maybe (fromMaybe)
 import Data.Monoid (Sum(..))
+import GHC.Magic (inline)
+#if __GLASGOW_HASKELL__ >= 802
+import GHC.Magic (noinline)
+#else
+#define noinline
```

Comment:
What is the purpose of this?

Reply:
it defines "noinline" to empty value. Wherever "noinline" occurs in the file it will be erased.

Reply:
Ah. Very sneaky.